Library/Formula/ddar.rb

```diff
@@ -9,7 +9,7 @@ class Ddar < Formula
 
   depends_on "xmltoman" => :build
   depends_on :python if MacOS.version <= :snow_leopard
-  depends_on "protobuf" => "with-python"
+  depends_on "protobuf"
```

Comment:
I don't know if we can express this properly with the current formula semantics. There doesn't seem to be a way of expressing "depends on X, where a particular option was *not*" used.

The `--with-python` option has no effect with `brew install protobuf`, since `install` just silently ignores any unrecognized options. When you do the `reinstall`, it remembers the previous `--without-python`. When you do a fresh `install protobuf --with-python`, it ignores `--with-python` and just goes with the defaults, which happens to be including python. To check, try using `--with-miss-piggy` instead of `--with-python`; it has the same effect. You can also do `brew install protobuf --without-python --with-python`; you'll end up without python.

Using an undefined option in the `depends_on` options list seems to have the effect of requiring the dependency to be built with the default options, which in this case works, and rebuilt from source every time instead of poured from a bottle. I think what happens is: since `--with-python` is undefined for `protobuf`, in `install ddar`, the dependency checker will check `protobuf`, see that `with-python` was not an option it was built with, do an installation of it using `--with-python`. The install step will ignore the unrecognized `--with-python`, so you end up with a build with the default options. Again, `depends_on "protobuf" => "with-miss-piggy"` has the same effect, so I think that's what's going on.

I looked through the source and doc, and it doesn't look like the formula code currently has a way of expressing a dependency on a formula option other than one which can be passed explicitly as an option.